### PR TITLE
Create initial set of specs for #can_castle?

### PR DESCRIPTION
**Pull request by radiofreemattd**
:warning:  :construction: :construction: :construction: :warning:
# UNDER CONSTRUCTION - WIP

## Task Overview
- [x] Verify with team that this represents a complete set of specs for the concept of `game` 's   `#can_castle?`
- [x] Solve issue in `king_spec` in which movement of the `king` generates an `ArgumentError` (*see* below for a further description of this problem).
- [x] Implement remainder of `#can_castle?` (requiring a check to see if any traversed space would result in the `king` being in check.
- [x] Write complete set of specs for `#castle!`

### CODE REVIEW

**Comment by radiofreemattd** [comment]
I've updated my original comment with a status update. Points to anyone who can point out my error as to why the expectation is not passing...

**Comment by radiofreemattd** [comment]
I believe this is now ready for review and to be patched in if @mscoutermarsh and @jonmast would take a look.

File: Gemfile [hunk 1]
```diff
@@ -46,11 +46,13 @@ group :development do
...
   # Adds spring compatibility to rspec
   gem 'spring-commands-rspec'
-  # Use Pry instead of IRB
-  gem 'pry-rails'
 end
 
 group :development, :test do
+  # Use Pry instead of IRB
+  gem 'pry-rails'
+  gem 'byebug'
```
Comment (jonmast) [inline Gemfile]: We could use pry-byebug instead which would be integrated with pry. I don't really care though as I doubt I will use this very often.

**Comment by jonmast** [comment]
:+1: Looks good

File: spec/models/king_spec.rb [hunk 2]
```diff
@@ -25,4 +25,151 @@
...
+    let(:white_kingside_rook)  { game.chess_pieces.find_by(position_x: 8, position_y: 1) }
+    let(:white_queenside_rook) { game.chess_pieces.find_by(position_x: 1, position_y: 1) }
+
+    it 'updates the x coordinates of the black king when castling kingside'do
+      remove_everything_but_rook_and_king!('black')
+      black_king.castle!(black_kingside_rook)
+      expect(black_king.reload.position_x).to eq 7
+    end
+
+    it 'updates the x coordinates of the black king when castling queenside'do
```
Comment (mscoutermarsh) [inline spec/models/king_spec.rb]: For these, I'd use a context block to try it up a bit.

```Ruby
context 'black' do
  before(:each) do
    remove_everything_but_rook_and_king!('black')
  end
  # etc etc
end
```

**Comment by mscoutermarsh** [comment]
love it! merge whenever you're ready!

**Pull request merged**